src/AspNet.Security.OAuth.Gitter/GitterAuthenticationHandler.cs

```diff
@@ -0,0 +1,93 @@
+﻿using AspNet.Security.OAuth.Extensions;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AspNet.Security.OAuth.Gitter
+{
+    public class GitterAuthenticationHandler : OAuthHandler<GitterAuthenticationOptions>
+    {
+        public GitterAuthenticationHandler([NotNull] HttpClient client)
+            : base(client)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue(tokens.TokenType, tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred when retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                    /* Status: */ response.StatusCode,
+                    /* Headers: */ response.Headers.ToString(),
+                    /* Body: */ await response.Content.ReadAsStringAsync());
+                throw new HttpRequestException("An error occurred when retrieving the user profile.");
+            }
+
+            var payload = JArray.Parse(await response.Content.ReadAsStringAsync());
+            var user = JObject.Parse(payload[0].ToString());
```

Comment:
Is this really necessary? Can't you just pick the `JObject` from the `JArray` instead of re-parsing it?

Something like `var user = (JObject) payload[0];`